### ISSUE TITLE
New rule: report absence of ARIA roles along side `epub:type`

### ISSUE THREAD

**Issue by rdeltour** [daisy/ace#70]
The DPUB ARIA roles have well-defined accessibility API mappings, and should be used in conjunction to `epub:type`.

Ace could include a rule to check that, and report as a "moderate" violation.

@mattgarrish thoughts welcome

**Comment by mattgarrish** [comment]
That would be useful, IMO.

We want people to stop thinking of epub:type as an accessibility mechanism, and the guidance is to prefer ARIA for accessibility but use both if you need some specific reading system functionality only supported by epub:type.

**Comment by rdeltour** [comment]
ok thanks, that's what I needed to know!

**Comment by laudrain** [comment]
Hi,
to start with, I don't see epub:type being reported in ace. Is it missing?

**Comment by rdeltour** [comment]
> I don't see epub:type being reported in ace. Is it missing?

You're right, it's not reported. 

To clarify, Ace will:
- report violations of some a11y guidelines (think WCAG + some EPUB-specific rules).
- extract some useful data from Content Documents, for later use in manual inspection.

@laudrain, are you saying some `epub:type`s should be extracted as useful data? which and why?

**Comment by laudrain** [comment]
Soem epub:type yes:

1) high level for logical structure: 
level 1: **frontmatter**
 and inside, level 2: **cover, tiltepage, copyright-page, ...**
level 1: **bodymatter**
 and inside, level 2: **part, chapter, subchapter, division**
level 1: **backmatter**
 and inside, level 2: **appendix, credits, index**

2) specifics: 
- **pagebreak** for page navigation checks
- **footnote, endnote** for notes in HTML aside

**Comment by rdeltour** [comment]
> Some epub:type yes

to be followed in #87...

**Comment by rdeltour** [comment]
_Moving out of Beta 2_
This rule can be quite disturbing at first for many people who've been told `epub:type` was good for accessibility, and will create many violation reports. I think it would be better to wait until we have an option to finely configure the rules run by Ace, so that people can easily disable this rule if it pollutes their report.

**Comment by mattgarrish** [comment]
Sorry, looks like I missed some notifications. I wouldn't make this an error even when implemented. WCAG only requires that user interface components use ARIA roles. This is more like a best practice for a best practice.
